generic3g/actions/MeanAction.F90

```diff
@@ -70,50 +85,30 @@ subroutine calculate_mean(this, rc)
 
    end subroutine calculate_mean
 
-   subroutine update_mean_accumulator(this, importState, exportState, clock, rc)
+   subroutine mean_update_pre(this, rc)
       class(MeanAction), intent(inout) :: this
-      type(ESMF_State) :: importState
-      type(ESMF_State) :: exportState
-      type(ESMF_Clock) :: clock
       integer, optional, intent(out) :: rc
 
       integer :: status
-      
-      _ASSERT(this%initialized(), 'Accumulator has not been initialized.')
-      if(.not. this%update_calculated) then
-         call this%calculate_mean(_RC)
-      end if
-      call this%AccumulatorAction%update(importState, exportState, clock, _RC)
-      _RETURN(_SUCCESS)
 
-   end subroutine update_mean_accumulator
-
-   subroutine invalidate_mean_accumulator(this, importState, exportState, clock, rc)
-      class(MeanAction), intent(inout) :: this
-      type(ESMF_State) :: importState
-      type(ESMF_State) :: exportState
-      type(ESMF_Clock) :: clock
-      integer, optional, intent(out) :: rc
-
-      integer :: status
-      
-      call this%AccumulatorAction%invalidate(importState, exportState, clock, _RC)
-      this%counter_scalar = this%counter_scalar + 1
+      call this%calculate_mean(_RC)
       _RETURN(_SUCCESS)
 
-   end subroutine invalidate_mean_accumulator
+   end subroutine mean_update_pre
 
    subroutine calculate_mean_R4(this, rc)
       class(MeanAction), intent(inout) :: this
       integer, optional, intent(out) :: rc
 
       integer :: status
       real(kind=ESMF_KIND_R4), pointer :: current_ptr(:) => null()
+      real(kind=ESMF_KIND_R4), pointer :: counter(:) => null()
```

Comment:
I thought we had agreed that `counter` would be I4 rather than R4?    Or is that to be deferred until later?

Reply:
I am working on the I4 right now. Parts of mapl3g_FieldPointerUtilities only support R4 and R8. I am extending them to support I4 (and I8 for good measure.)

Reply:
Thanks.   My memory is just poor so I needed that confirmation.